department-of-veterans-affairs/caseflow
# Updates to EditContentionTitle

Connects #14994

### Description
This rewrites `EditContentionTitle` as a functional component, and adds Jest tests and Storybook stories. This rewrite also fixes the issue whereby state would not get updated even if the issue prop did, which meant that the initial value displayed in the textarea would sometimes not match the issue with which it was associated (if another issue had been removed).

### Acceptance Criteria
- [ ] Value in textarea is set correctly when opened, even if other issues in list have been removed

### Testing Plan
1. Add `binding.pry` to line 1594 of spec/feature/intake/higher_level_review/edit_spec.rb
2. ```bundle exec rspec spec/feature/intake/higher_level_review/edit_spec.rb:1594```
3. Add another issue (either one, and say "Yes" to request extension)
4. Click "Edit Contention Title" on the new issue you added
5. Verify the value in `textarea` is correct
6. Remove one of the two (noneditable) apportionment ratings up above
7. Edit the new issue again
  - on master the `textarea` will have value of the other editable issue ("Right Knee")
  - with this fix, that should show the correct value
